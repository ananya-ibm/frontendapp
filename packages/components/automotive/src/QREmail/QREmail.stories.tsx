/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { QREmail } from './QREmail';

export default {
  title: 'Components/Core/QREmail',
  component: QREmail
};

export const normal = args => <QREmail {...args} />;
normal.args = {};
