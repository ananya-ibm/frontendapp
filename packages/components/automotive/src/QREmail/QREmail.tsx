/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import { ButtonGroup, TextInput } from '@exo/frontend-components-base';
import { SecondaryButton } from '@carbon/react';
import React, { useState } from 'react';
import * as S from './QREmail.styles';

// TODO: Move this component

export const QREmail = ({ onSubmit = () => {} }: Props) => {
  const [email, setEmail] = useState('');
  const handleSubmit = (e) => {
    e.preventDefault();
    onSubmit(email);
  };

  return (
    <S.QREmail onSubmit={handleSubmit}>
      <ButtonGroup>
        <SecondaryButton onClick={() => {}}>Send QR Code on email</SecondaryButton>
        <TextInput
          id="email"
          name="qremail"
          placeholder="Enter email"
          labelText="Your email"
          hideLabel
          type="email"
          value={email}
          onChange={(e) => {
            setEmail(e.target.value);
          }}
        />{' '}
      </ButtonGroup>
    </S.QREmail>
  );
};

type Props = {
  buttonText?: string | React.ReactNode;
  onSubmit?: (email?: string) => void;
};
