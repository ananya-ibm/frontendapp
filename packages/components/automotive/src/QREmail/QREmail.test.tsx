/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { screen, fireEvent } from '@testing-library/react';
import { renderWithTheme as render } from '@testUtils';
import userEvent from '@testing-library/user-event';
import { axe } from 'jest-axe';
import { QREmail } from './QREmail';

describe('<QREmail /> component', () => {
  test('its snapshot matches', () => {
    const { container } = render(<QREmail />);

    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('clicking submit invokes function that returns email', async () => {
    const mockFn = jest.fn(values => {
      return values;
    });
    render(<QREmail onSubmit={mockFn} />);

    const input = screen.getByLabelText(/your email/i);
    const submit = screen.getByText(/QR/i);

    await userEvent.type(input, 'email@email.com');
    fireEvent.click(submit);
    //need to fix testing
    //expect(mockFn.mock.calls.length).toBe(1);
    //expect(mockFn.mock.results[0].value.toString()).toBe('email@email.com');
  });
  test('it passes an axe accesibility check', async () => {
    const { container } = render(<QREmail />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
