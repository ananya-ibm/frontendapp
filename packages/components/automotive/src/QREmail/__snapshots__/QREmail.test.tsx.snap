// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QREmail /> component its snapshot matches 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  gap: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 40rem;
}

@media (min-width:672px) {
  .c1 {
    -webkit-flex-wrap: no-wrap;
    -ms-flex-wrap: no-wrap;
    flex-wrap: no-wrap;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
     
    <div
      class="cds--form-item cds--text-input-wrapper"
    >
      <label
        class="cds--label cds--visually-hidden"
        for="email"
      >
        Your email
      </label>
      <div
        class="cds--text-input__field-outer-wrapper"
      >
        <div
          class="cds--text-input__field-wrapper"
        >
          <input
            class="cds--text-input cds--text__input cds--text-input--md"
            id="email"
            name="qremail"
            placeholder="Enter email"
            title="Enter email"
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
    <button
      class="cds--btn cds--btn--secondary"
      tabindex="0"
      type="button"
    >
      Send QR Code on email
    </button>
  </div>
</div>
`;
